src/general/PauseMenu.cs

```diff
@@ -70,24 +70,34 @@ public override void _Ready()
         saveMenu = GetNode<NewSaveMenu>(SaveMenuPath);
     }
 
-    [RunOnKeyDown("ui_cancel")]
-    public void EscapeKeyPressed()
+    /// <summary>
+    ///   Priority lower to avoid handling SubMenu ESCs.
```

Comment:
Shouldn't this be the other way around? I like that you can quickly close the entire pause menu by pressing ESC.

Reply:
Weird if you can close the entire pause menu when you are in other submenus but can only go back to pause menu if you are in Options menu. So I unified it.

Reply:
I just tested, when in the pause menu, pressing ESC in any of the sub menus, directly resumes the game:
- Save Game
- Load Game
- Help
- Options

I think this version, will make things worse as the Options menu will now just go back to the pause menu instead of resuming the game.

Also I just playtested this in general and noticed a problem: when pressing ESC the button press sound plays, when it shouldn't.

Reply:
Well, that's because I called some button call backs which played button sounds. I was wondering that if pressing escape should always make the button sound play. But if I don't directly call button callback, I have to duplicate some codes which is also not expected to happen.

As for the first paragraph, I didn't really understand. In my opinion, always go back to the previous layer of menu is a good design.

Reply:
I want to preserve how the pause menu currently works on the master branch: pressing ESC always unpauses the game.

For the sounds, if it's actually necessary (and can't be avoided with the design on the main menu code just hiding the sub menus when pressing ESC, other than the options menu), then create a separate method that closes the menu and one that closes the menu from a button press and plays the sound.